runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/BoundedQueueExecutor.java

```diff
@@ -115,12 +143,16 @@ public boolean executorQueueIsEmpty() {
     return executor.getQueue().isEmpty();
   }
 
-  public long allThreadsActiveTime() {
+  public synchronized long allThreadsActiveTime() {
     return totalTimeMaxActiveThreadsUsed;
   }
 
-  public int activeCount() {
-    return activeCount.intValue();
+  public synchronized int activeCount() {
+    return activeCount;
+  }
+
+  public synchronized int maximumThreadCount() {
+    return maximumThreadCount;
   }
 
   public long bytesOutstanding() {
```

Comment:
doesn't look modified by this PR but bytesOutstanding() and elementsOutstanding() don't seem safe to read here if they are not beneath the monitor

can these public methods be removed? or shoudl they be changed to use monitor?

Reply:
Yeah they are not thread safe. They are currently used in StreamingDataflowWorker to gather metrics so it seems okay to me to leave them unsafe but in case we use them for other purposes, I feel it is better to be changed to use monitor.